### PR TITLE
MRG: Add NIRS support to animate topomap

### PR DESCRIPTION
**Pull request by rob-luke**
#### What does this implement/fix?
Add support for NIRS types to `animate_topomap`.

#### Additional information
I am not happy with the test I have added. See below. Suggestions welcomed.

### CODE REVIEW

File: mne/viz/tests/test_topomap.py [hunk 1]
```diff
@@ -136,6 +140,19 @@ def test_plot_topomap_animation():
...
+    raw = read_raw_nirx(fname_nirx_15_0, preload=True)
+    raw = nirs.optical_density(raw)
+    raw = nirs.beer_lambert_law(raw)
+    events, _ = events_from_annotations(raw, event_id={'1.0': 1, '2.0': 2})
+    event_dict = {'A/b': 2, 'A/c': 1}
+    epochs = Epochs(raw, events, event_id=event_dict, tmin=-1, tmax=2)
+    evoked = epochs['A'].average()
+    fig, anim = evoked.animate_topomap(ch_type='hbo')
+    anim._func(1)  # _animate has to be tested separately on 'Agg' backend.
+    assert len(fig.axes) == 2
```
Comment (rob-luke) [inline mne/viz/tests/test_topomap.py]: Originally I tried to change the file types of the evoked variable to hbo for this test. But due to a projection and average referencing I was unable to use the same approach as in my previous PR (see below). This code seems excessive, can anyone suggest a nicer way of testing this?

https://github.com/mne-tools/mne-python/pull/7001/files#diff-95830e06fd39b594b34b28b6fecf7296R424-R428

Comment (agramfort) [reply mne/viz/tests/test_topomap.py]: just create fake data with EvokedArray and create_info + set_montage. clear?

Comment (rob-luke) [reply mne/viz/tests/test_topomap.py]: Thanks @agramfort for the point in the right direction. I am now happy with this PR, could you please review.

**Comment by larsoner** [comment]
@rob-luke is it worth adding or updating some `latest.inc` entry? Maybe just tack your name and the `:func:` or `:meth:` onto https://github.com/mne-tools/mne-python/blob/master/doc/changes/latest.inc#L38 ? That way all the viz stuff can be grouped together in one entry

**Comment by rob-luke** [comment]
#7057

**Comment by rob-luke** [comment]
> @rob-luke is it worth adding or updating some `latest.inc` entry? Maybe just tack your name and the `:func:` or `:meth:` onto https://github.com/mne-tools/mne-python/blob/master/doc/changes/latest.inc#L38 ? That way all the viz stuff can be grouped together in one entry

Thanks

**Comment by agramfort** [comment]
@larsoner merge when happy

>

**Comment by larsoner** [comment]
Thanks @rob-luke

**Pull request merged**